docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,277 @@
+
+- Fresh Firecracker microVMs are booted using `anonymous` memory, while microVMs
+  resumed from snapshot load memory on-demand from the snapshot and copy-on-write
+  to anonymous memory.
+- Resuming from a snapshot is optimized for speed. Taking a snapshot always involves
+  synchronously writing dirty memory pages to the memory snapshot file. Taking a
+  snapshot of a fresh microVM running off non-file-backed anonymous memory, results in
+  full contents of guest memory being written to the snapshot.
+- The snapshot operations have minimal requirements/dependencies on when/how they
+  should be used.
+- The API calls exposing the snapshotting functionality have clear **Prerequisites** that
+  describe the previously mentioned minimal requirements.
+- The _memory file_ and _microVM state file_ are generated by Firecracker on snapshot
+  creation and the disk contents are flushed to their backing files.
+ 
+## Snapshot versioning
+
+Firecracker snapshotting implementation offers support for microVM versioning
+(`cross-version snapshots`) in the following contexts:
+- saving snapshots at older versions (also called `rollback`),
+- loading snapshots from older versions (also called `live update`).
+Live update refers to successfully loading any snapshot created by Firecracker version
+`N` in a Firecracker version `N+o`.
+Rollback means being able to create a snapshot with any version in `[N, N + o]` interval
+while being in Firecracker version `N+o`.
+
+The value of `o` (maximum number of older versions that we can restore from / save
+a snapshot to, from the current version) is not yet decided, but we plan on offering
+backwards compatibility for at least one year changes. 
+
+## Creating snapshots
+
+The supported and of interest operations in the snapshotting context are: `Pause`,
+`Resume`, `CreateSnapshot` and `LoadSnapshot`. `Pause`, `Resume` and `CreateSnapshot`
+can be called only after booting the microVM, while `LoadSnapshot` is allowed only
+before boot.
+
+For creating a snapshot, you have to pause the running microVM, which means pausing
+the microVM' vCPUs. If you change your mind and want to create a snapshot later,
+you can resume the microVM and then pause it again when you decide to take that
+snapshot.
+
+For pausing the microVM you should send the following API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PATCH "http://localhost/vm" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+            "state": "Paused"
+    }"
+```
+
+**Prerequisites**: You can pause the microVM any time after the microVM is booted.
+                   Successive calls of this request keep the microVM in the `Paused`
+                   state.
+**Effects**:
+- _on success_: microVM is guaranteed to be `Paused`.
+- _on failure_: no side-effects.
+
+Now that the microVM is paused, you can create a snapshot, which can be either a `full`
```

Comment:
Full vs diff snapshots are a big enough thing to deserve their own chapter, explaining their utility in various use cases, and how to use diffs.

Reply:
I made the separation a little bit clearer.